server/routes/journeys/referral/check-answers/tests.cy.ts

```diff
@@ -0,0 +1,306 @@
+import { v4 as uuidV4 } from 'uuid'
+import { injectJourneyDataAndReload } from '../../../../../integration_tests/utils/e2eTestUtils'
+import { checkAxeAccessibility } from '../../../../../integration_tests/support/accessibilityViolations'
+
+context('test /record-investigation/check-answers', () => {
+  const uuid = uuidV4()
+  const START_URL = `${uuid}/prisoners/A1111AA/referral/start`
+  const PAGE_URL = `${uuid}/referral/check-answers`
+  const ESCAPE_LITTLE = '<script>alert("Test User")</script>'
+  const ESCAPE_AND_FORMAT_EXPECTED =
+    "\n      Text\n      \n        • Bullet 1\n        • Bullet 2\n        • Bullet 3\n        \n        Paragraph\n        \n        <script>alert('xss');</script>\n        \n        <button>also should be escaped</button>\n    "
+  const ESCAPE_AND_FORMAT = `Text
+
+  • Bullet 1
+  • Bullet 2
+  • Bullet 3
+  
+  Paragraph
+  
+  <script>alert('xss');</script>
+  
+  <button>also should be escaped</button>`
+
+  const getContinueButton = () => cy.findByRole('button', { name: /continue/i })
+
+  beforeEach(() => {
+    cy.task('reset')
+    cy.task('stubSignIn')
+    cy.task('stubComponents')
+    cy.task('stubAreaOfWork')
+    cy.task('stubIncidentLocation')
+    cy.task('stubIncidentType')
+    cy.task('stubIntervieweeRoles')
+    cy.task('stubDecisionSignerRoles')
+    cy.task('stubContribFactors')
+    cy.task('stubIncidentInvolvement')
+    cy.task('stubOutcomeType')
+    cy.task('stubGetPrisonerImage')
+    cy.task('stubGetPrisoner')
+
+    cy.signIn()
+    cy.visit(START_URL)
+
+    injectJourneyDataAndReload(uuid, {
+      referral: {
+        assaultedStaffName: 'Jimmy',
+        contributoryFactors: [{ factorType: { code: 'CODE1', description: 'Factor1' } }],
+        descriptionOfConcern: 'idklol',
+        incidentDate: '22 August 2024',
+        incidentInvolvement: { code: 'A', description: 'TypeA' },
+        incidentLocation: { code: 'A', description: 'LocationA' },
+        incidentType: { code: 'A', description: 'IncidentA' },
+        incidentTime: '01:01',
+        isOnBehalfOfReferral: false,
+        knownReasons: 'idklol',
+        refererArea: { code: 'A', description: 'AreaA' },
+        referredBy: 'Test Person',
+        staffAssaulted: false,
+      },
+    })
+
+    cy.visit(PAGE_URL)
+
+    cy.url().should('to.match', /\/check-answers/)
+    checkAxeAccessibility()
+  })
+
+  it('should redirect to something on failure', () => {
+    cy.task('stubCsipRecordPostFailure')
+    cy.findByRole('button', { name: /Confirm and send/i }).click()
+    cy.url().should('to.match', /\/check-answers/)
+    cy.findByText('There is a problem').should('be.visible')
+  })
+
+  it('should be able to change answers and proceed to confirmation', () => {
+    cy.task('stubCsipRecordPostSuccess')
```

Comment:
Can we have this split up into discrete functions like the other longer cypress tests are, just so it's more immediately understandable?